salalib/shapemap.cpp

```diff
@@ -1943,8 +1945,8 @@ Point2f ShapeMap::getClosestVertex(const Point2f& p) const
    PixelRef pix = pixelate(p);
 
    double mindist = -1.0;
-   for (size_t i = 0; i < m_pixel_shapes[pix.x][pix.y].size(); i++) {
-      ShapeRef& ref = m_pixel_shapes[pix.x][pix.y][i];
+   for (size_t i = 0; i < m_pixel_shapes[pix.x + pix.y*m_cols].size(); i++) {
+      const ShapeRef& ref = m_pixel_shapes[pix.x + pix.y*m_cols][i];

```

Comment:
Hard to see all the code this is here in github, but do we need to do this with indices? At least in a few instances, I think an `for(auto& shape: currentShape)` might actually do.